js/app/src/lite/Playground.svelte

```diff
@@ -0,0 +1,232 @@
+	export let initial_height: string;
+	export let app_mode: boolean;
+	export let is_embed: boolean;
+	export let theme_mode: ThemeMode | null = "system";
+	export let control_page_title: boolean;
+	export let container: boolean;
+	export let info: boolean;
+	export let eager: boolean;
+	export let mount_css: typeof default_mount_css = default_mount_css;
+	export let client: ReturnType<typeof api_factory>["client"];
+	export let upload_files: ReturnType<typeof api_factory>["upload_files"];
+	export let worker_proxy: WorkerProxy | undefined = undefined;
+	export let fetch_implementation: typeof fetch = fetch;
+	export let EventSource_factory: (url: URL) => EventSource = (url) =>
+		new EventSource(url);
+	export let space: string | null;
+	export let host: string | null;
+	export let src: string | null;
+
+	export let code: string | undefined;
+	export let error_display: SvelteComponent | null;
+
+	const dispatch = createEventDispatcher();
+	// $: dispatch("code", { code });
+
+	let dummy_elem: any = { classList: { contains: () => false } };
+	let dummy_gradio: any = { dispatch: (_) => {} };
+
+	let loading_text = "";
+	export let loaded = false;
+	worker_proxy?.addEventListener("progress-update", (event) => {
+		loading_text = (event as CustomEvent).detail + "...";
+		if (loading_text === "Initialization completed...") {
+			loaded = true;
+		}
+	});
+
+	$: loading_text;
+	$: loaded;
+</script>
+
+<svelte:head>
+	<link rel="stylesheet" href="https://gradio-hello-world.hf.space/theme.css" />
+</svelte:head>
+
+<div class="p-4 w-full h-full">
+	<div
+		class="flex flex-col md:flex-row w-full h-full border border-gray-200 rounded-md"
+	>
+		<div class:md:border-r={loaded} class="code-editor grow flex-1">
+			<div class="flex justify-between align-middle h-8 border-b pl-4 pr-2">
+				<h3 class="pt-1 grow">Code</h3>
+				{#if !loaded}
+					<div class="flex"></div>
+					<div class="flex float-right items-center" style="color:#68696b">
+						<div class="loading-dot mx-2"></div>
+						{loading_text}
+					</div>
+				{:else}
+					<div class="flex items-center mx-2" style="color:#68696b">
+						<div class="loaded-dot mx-2"></div>
+						Interactive
+					</div>
+					<button
+						class="bg-gray-200 text-gray-500 font-bold px-2 rounded float-right m-1"
+						style="
+                        background: #eaecef;
+                        color: #374151;
+                        font-weight: 500;
+
+                        "
+						on:click={() => {
+							dispatch("code", { code });
+						}}
+					>
+						Update →
+					</button>
+				{/if}
+			</div>
+			{#if loaded}
+				<Code
+					bind:value={code}
+					label=""
+					language="python"
+					target={dummy_elem}
+					gradio={dummy_gradio}
+					lines={10}
+					interactive={true}
+				/>
+			{:else}
+				<Code
+					bind:value={code}
+					label=""
+					language="python"
+					target={dummy_elem}
+					gradio={dummy_gradio}
+					lines={10}
+					interactive={false}
+				/>
+			{/if}
```

Comment:
This doesn't work actually, and I'm not sure why. But it doesn't make the code component interactive after loaded=true.

Reply:
The reason this is a problem is because `interactive` isn't interactive (oh the irony).

We set the 'interactivity' by changing the `readonly` status of the codemirror instance, which is controlled via an extension (just codemirror things, everything is an extension).

The fix for this is weird but we would need to do something like this (untested but i think should work with small tweaks):

```ts
const _readonly = new Compartment();
const _editable = new Compartment();

// when we set the initial extension state (around line 200)

const extensions: Extension[] = [
  _readonly.of(EditorState.readOnly.of(readonly)),
  _editable.of(EditorView.editable.of(!readonly)),
  EditorView.contentAttributes.of({ "aria-label": "Code input container" })
];


// when the readonly prop changes

function set_readonly(view: EditorView, readonly: boolean): void {
  view.dispatch({
    effects: [
        _readonly.reconfigure(EditorState.readOnly.of(readonly)),
        _editable.reconfigure(EditorView.editable.of(!readonly))
    ]
  });
}

$: set_readonly(view, readonly);
```

If you want you could merge this as is and we can fix this in a followup, not a huge deal.